dev/benchmarking/ddl/1_11_table_public_benchmark.sql

```diff
@@ -0,0 +1,58 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+*/
+
+CREATE EXTENSION IF NOT EXISTS "citext";  -- type for case-insensitive text
+
+-- BENCHMARK
+CREATE TABLE IF NOT EXISTS public.benchmark
+(
+  benchmark_id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , benchmark_name citext NOT NULL
+  , parameter_names text[]
+  , lessisbetter boolean NOT NULL
+  , benchmark_description text NOT NULL
+  , benchmark_version citext NOT NULL
+  , unit_id integer NOT NULL
+  , PRIMARY KEY (benchmark_id, benchmark_language_id)
```

Comment:
This gets into the weeds a little. In order to easily error-check that `benchmark_language_id` is the same one from both the `benchmark` table and the `environment` table when creating an entry in `benchmark_run`, the ID must be pulled from both tables as a Foreign Key.

This was the cleanest way I could think of to do this without writing custom error checks.

You are not the first to note that this is confusing; @emkornfield also noted this, and his proposed solution was to take the language completely out of the benchmark definition and have it exclusively reside in `environment`. I have no opinion because I am not 100% sure what the purpose of benchmarks in different languages are (per comments by @aregm) but am happy to implement whatever other people say since benchmarking is outside of my skillset. (database-ing is something I can contribute, though!)

Reply:
Ah, fair enough. I think we can keep it like that then. Do you know if that might cause issues with other utilities wanting to access the DB, say Django's ORM? And would there be an easy way to migrate out of it?

Reply:
> Do you know if that might cause issues with other utilities wanting to access the DB, say Django's ORM?

It is not a problem; the complexity in the underlying data model is hidden by the views, if I have done my job.

> And would there be an easy way to migrate out of it?

- It is do-able for sure thanks to the existing unique `benchmark_id`.
- All of the views and trigger functions can just be deleted and replaced
- The only pain point would be the downstream tables.
I have never tried unlinking a foreign key and re-defining it, but at worst case the tables can be temporarily copied to ones with a different name and re-inserted into the newly defined tables. (The total number of rows should be small for `benchmark` definitions and for the other tables that do this (`environment`, `language_dependencies`, `language_implementation_version`))